tools/clang/lib/SPIRV/DeclResultIdMapper.cpp

```diff
@@ -234,8 +234,6 @@ SpirvEvalInfo DeclResultIdMapper::getDeclResultId(const NamedDecl *decl) {
       return *info;
     }
 
-  assert(false && "found unregistered decl");
```

Comment:
I think removing this assert makes the code somewhat loose in terms of checks.
What I would do is change the function to
```cpp
SpirvEvalInfo getDeclResultId(const NamedDecl *decl, bool declMustExist = true);
```
and then
```cpp
if(declMustExist) {
  assert(false && "found unregistered decl");
  decl->dump();
}
```

Then, inside `doDeclRefExpr` function:
```cpp
auto id = declIdMapper.getDeclResultId(decl, /*declMustExist*/ false);
```
And all other calls to `getDeclResultId` will get the assertion by default, which is a good thing IMHO.

Reply:
Yep. However, `assert` will be gone completely for release build. So I used `emitFatalError` instead. So that way we still notify the devs something is wrong, which I intended to do for a long time.